src/types.ts

```diff
@@ -573,20 +574,37 @@ export class Signature {
   cachedFunctionTarget: FunctionTarget | null = null;
   /** Respective function type. */
   type: Type;
+  /** The program that created this signature. */
+  program: Program;
 
   /** Constructs a new signature. */
   constructor(
+    program: Program,
     parameterTypes: Type[] | null = null,
     returnType: Type | null = null,
-    thisType: Type | null = null
+    thisType: Type | null = null,
   ) {
     this.parameterTypes = parameterTypes ? parameterTypes : [];
     this.parameterNames = null;
     this.requiredParameters = 0;
     this.returnType = returnType ? returnType : Type.void;
     this.thisType = thisType;
+    this.program = program;
     this.hasRest = false;
     this.type = Type.u32.asFunction(this);
+
+    let compare: Signature;
+    let signatureTypes = program.uniqueSignatures;
+    let length = signatureTypes.length;
+    for (let i = 0; i < length; i++) {
+      compare = signatureTypes[i];
```

Comment:
nit: `let compare =`, and remove the var statement above

Reply:
Not a problem. This was left over from an initial implementation that required the Signature to stick around after the loop. Easy fix. Thanks for the heads up.

Reply:
Should be all set.